### PR TITLE
OAK-8967: OR  query with ORDER BY don't work as expected

### PR DESCRIPTION
**Pull request by tihom88**

### CODE REVIEW

**Review by thomasmueller — APPROVED** [review]

File: oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+public class OrQueryOrderLimitWithoutIndexTest extends AbstractQueryTest {
+
+    private final static String[] baseNodePaths = {"UnionQueryTest", "UnionQueryTest1"};
```
Comment (fabriziofortino) [inline oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java]: nit: this should be UPPERCASE

**Review by fabriziofortino — APPROVED** [review]
LGTM

File: oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        Result result = qe.executeQuery(query, QueryEngineImpl.SQL2, limit, offset,
+                QueryEngine.NO_BINDINGS, QueryEngine.NO_MAPPINGS);
+
+        List<ResultRow> rows = Lists.newArrayList(result.getRows());
+        Assert.assertEquals(expected.length, rows.size());
+
+        int i = 0;
+        for (ResultRow rr : result.getRows()) {
+            System.out.println(rr.getPath());
```
Comment (averma21) [inline oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java]: did this print statement slip through or would you like to have path printed?

**Review by averma21 — APPROVED** [review]

File: oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java (hunk 1 above)
Comment (tihom88) [reply oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java]: Updated

File: oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java (hunk 2 above)
Comment (tihom88) [reply oak-core/src/test/java/org/apache/jackrabbit/oak/query/OrQueryOrderLimitWithoutIndexTest.java]: Removed.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java [hunk 3]
```diff
@@ -1470,6 +1470,7 @@ private UnionQueryImpl newAlternativeUnionQuery(@NotNull Query left, @NotNull Qu
         u.setMeasure(measure);
         u.setInternal(isInternal);
         u.setQueryOptions(queryOptions);
+        u.setOrderings(this.getOrderings());
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java]: This is fine, or (for consistency): 

    u.setOrderings(orderings);

**Review by thomasmueller — APPROVED** [review]

File: oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java (hunk 3 above)
Comment (tihom88) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java]: Done

**Comment by tihom88** [comment]
Merged

**Pull request closed without merging**